File: backend/app/schemas/progress.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel, ConfigDict

class ProgressCreateRequest(BaseModel):
    exercise_name: str
    workout_type: Optional[str] = None
    duration_minutes: int
    reps_completed: int
    calories_burned: float
    posture_score: float
    completion_percentage: float
    notes: Optional[str] = None

class ProgressResponse(ProgressCreateRequest):
    id: int
    user_id: int
    workout_date: datetime
    streak_day: int
    created_at: datetime
    
    model_config = ConfigDict(from_attributes=True)

class DailyStats(BaseModel):
    date: str
    workouts: int
    calories: float
    avg_score: float

class ProgressStatsResponse(BaseModel):
    total_workouts: int
    total_minutes: int
    total_calories: float
    current_streak: int
    best_streak: int
    best_posture_score: float
    weekly_history: List[DailyStats]
    recent_logs: List[ProgressResponse]

File: frontend/src/api/progress.ts
export interface ProgressCreateRequest {
  exercise_name: string;
  workout_type?: string;
  duration_minutes: number;
  reps_completed: number;
  calories_burned: number;
  posture_score: number;
  completion_percentage: number;
  notes?: string;
}

export interface ProgressResponse extends ProgressCreateRequest {
  id: number;
  user_id: number;
  workout_date: string;
  streak_day: number;
}

export interface DailyStats {
  date: string;
  workouts: number;
  calories: number;
  avg_score: number;
}

export interface ProgressStatsResponse {
  total_workouts: number;
  total_minutes: number;
  total_calories: number;
  current_streak: number;
  best_streak: number;
  best_posture_score: number;
  weekly_history: DailyStats[];
  recent_logs: ProgressResponse[];
}

export const logProgress = async (data: ProgressCreateRequest): Promise<ProgressResponse> => {
  const token = localStorage.getItem("access_token");
  const res = await fetch("http://127.0.0.1:8000/api/v1/progress/log", {
    method: "POST",
    headers: {
      "Content-Type": "application/json",
      Authorization: `Bearer ${token}`
    },
    body: JSON.stringify(data)
  });
  if (!res.ok) throw new Error("Failed to log progress");
  return res.json();
};

export const getProgressStats = async (): Promise<ProgressStatsResponse> => {
  const token = localStorage.getItem("access_token");
  const res = await fetch("http://127.0.0.1:8000/api/v1/progress/stats", {
    headers: { Authorization: `Bearer ${token}` }
  });
  if (!res.ok) throw new Error("Failed to load progress stats");
  return res.json();
};
